operator/internal/controller/redpanda/role_controller.go

```diff
@@ -51,10 +50,15 @@ func (r *RoleReconciler) FinalizerPatch(request ResourceRequest[*redpandav1alpha
 
 func (r *RoleReconciler) SyncResource(ctx context.Context, request ResourceRequest[*redpandav1alpha2.RedpandaRole]) (client.Patch, error) {
 	role := request.object
-	hasManagedACLs, hasManagedRole := role.HasManagedACLs(), role.HasManagedRole()
-	shouldManageACLs, shouldManageRole := role.ShouldManageACLs(), role.ShouldManageRole()
+	// SyncResource ensures the role exists, applies ACLs if Authorization is set,
+	// and synchronizes inline principals from spec.principals. Aggregation from
+	// RoleBindings is not yet implemented.
+	hasManagedACLs, hasManagedRole, hasManagedPrincipals := role.HasManagedACLs(), role.HasManagedRole(), role.HasManagedPrincipals()
+	shouldManageACLs, shouldManageRole, ShouldManagePrincipals := role.ShouldManageACLs(), role.ShouldManageRole(), role.ShouldManagePrincipals()
```

Comment:
NIT: I would use a lowercase letter for the first character of the variable name.
```suggestion
	shouldManageACLs, shouldManageRole, shouldManagePrincipals := role.ShouldManageACLs(), role.ShouldManageRole(), role.ShouldManagePrincipals()
```